src/NuGet.Services.AzureSearch/Db2AzureSearch/Db2AzureSearchCommand.cs

```diff
@@ -134,6 +127,41 @@ private async Task InitializeAsync()
             await _indexBuilder.CreateHijackIndexAsync();
         }
 
+        private async Task PushAllPackageRegistrationsAsync(CancellationTokenSource cancelledCts, CancellationTokenSource produceWorkCts, ConcurrentBag<IdAndValue<IReadOnlyList<string>>> allOwners)
+        {
+            var allWork = new ConcurrentBag<NewPackageRegistration>();
+            var producerTask = ProduceWorkAsync(allWork, produceWorkCts, cancelledCts.Token);
+            var consumerTasks = Enumerable
+                .Range(0, _options.Value.MaxConcurrentBatches)
+                .Select(i => ConsumeWorkAsync(allWork, allOwners, produceWorkCts.Token, cancelledCts.Token))
+                .ToList();
+            var allTasks = new[] { producerTask }.Concat(consumerTasks).ToList();
+
+            // If one of the tasks throws an exception before the work is completed, cancel the work.
+            var firstTask = await Task.WhenAny(allTasks);
+            if (firstTask.IsFaulted)
+            {
+                cancelledCts.Cancel();
+            }
+
+            await firstTask;
+            await Task.WhenAll(allTasks);
+        }
+
+        private async Task WriteOwnerDataAsync(ConcurrentBag<IdAndValue<IReadOnlyList<string>>> allOwners)
+        {
+            _logger.LogInformation("Building and writing the initial owners file.");
+            var ownersBuilder = new PackageIdToOwnersBuilder(_logger);
+            foreach (var owners in allOwners)
+            {
+                ownersBuilder.Add(owners.Id, owners.Value);
+            }
+
+            await _ownerDataClient.ReplaceLatestIndexedAsync(
+                ownersBuilder.GetResult(),
+                AccessConditionWrapper.GenerateIfNotExistsCondition());
+        }
```

Comment:
Consider logging when the work completes.

Reply:
This is handled by the job infrastructure.

Reply:
Oh I meant at the granularity of generating the owners.v2.json file. Considering it's a big file, this may be slower than expected.

Reply:
Oh okay, I'll add this in my current branch for the next set of changes.